engine/src/flutter/impeller/renderer/backend/vulkan/pipeline_library_vk.cc

```diff
@@ -179,7 +172,8 @@ PipelineFuture<PipelineDescriptor> PipelineLibraryVK::GetPipeline(
 
   auto weak_this = weak_from_this();
 
-  auto generation_task = [descriptor, weak_this, promise]() {
+  uint64_t next_key = pipeline_key_++;
```

Comment:
Is `GetPipeline` guaranteed to be called from the same thread?

Reply:
I added a guard that verifies that is synchronized by the same mutex that guards the pipeline map.